source/common/protobuf/utility.cc

```diff
@@ -88,6 +88,21 @@ ProtoValidationException::ProtoValidationException(const std::string& validation
   ENVOY_LOG_MISC(debug, "Proto validation error; throwing {}", what());
 }
 
+void MessageUtil::checkUnknownFields(const Protobuf::Message& message,
+                                     ProtobufMessage::ValidationVisitor& validation_visitor) {
+  auto& unknown_fields = message.GetReflection()->GetUnknownFields(message);
```

Comment:
Should this be done recursively for embedded message fields?
If I remember, this method also causes allocation of metadata and has performance impact. Perhaps, things are better now?

Reply:
This is a fair point; this is actually a bug in the existing implementation. We should do it (folks can always disable the checking if they don't want the performance impact). Let me fix that in a followup PR, tracking in https://github.com/envoyproxy/envoy/issues/7980.

Reply:
nit: const auto